### PR TITLE
Fix some bugs: Alessio

### PR DESCRIPTION
**Pull request by AlessioVerardo**
In this PR, 
I fixed some bugs like: 
- The spinner on the EditProfileActivity
- The fact that we can add profile with an empty name

I also add scrollable view on the different home fragment for each different provider so that even small screen can click on all the buttons. I also add a function to generate an alert dialog in the helper functions ! 

The test will come tomorrow :)

### CODE REVIEW

File: app/src/main/res/layout/activity_edit_profile.xml [hunk 1]
```diff
@@ -70,13 +70,13 @@
                     android:id="@+id/EditProfileActivity_Rank"
                     android:layout_width="match_parent"
                     android:layout_height="wrap_content"
+                    android:editable="false"
                     android:layout_gravity="center_vertical"
                     android:drawableStart="@drawable/ic_baseline_upgrade_24"
                     android:drawablePadding="10dp"
                     android:paddingStart="20dp"
                     android:paddingEnd="0dp"
```
Comment (Mathematou) [inline app/src/main/res/layout/activity_edit_profile.xml:78]: Is it really useful to define the padding to 0? Is it not by default?

**Review by Mathematou — APPROVED** [review]
Nice code as always :)

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/EditProfileActivity.kt [hunk 2]
```diff
@@ -123,15 +127,15 @@ class EditProfileActivity: AppCompatActivity() {
...
-            UserRole.ORGANIZER -> return resources.getStringArray(R.array.Ranks)[1]
-            UserRole.STAFF -> return resources.getStringArray(R.array.Ranks)[2]
-            UserRole.PARTICIPANT -> return resources.getStringArray(R.array.Ranks)[3]
+        return when (rank) {
+            UserRole.ADMIN -> resources.getStringArray(R.array.Ranks)[0]
+            UserRole.ORGANIZER -> resources.getStringArray(R.array.Ranks)[1]
+            UserRole.STAFF -> resources.getStringArray(R.array.Ranks)[2]
+            UserRole.PARTICIPANT -> resources.getStringArray(R.array.Ranks)[3]
         }
     }
```
Comment (Rygone) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/activity/EditProfileActivity.kt:136]: can be written like this
```
fun rankToString(rank: UserRole) =
        when (rank) {
            UserRole.ADMIN -> resources.getStringArray(R.array.Ranks)[0]
            UserRole.ORGANIZER -> resources.getStringArray(R.array.Ranks)[1]
            UserRole.STAFF -> resources.getStringArray(R.array.Ranks)[2]
            UserRole.PARTICIPANT -> resources.getStringArray(R.array.Ranks)[3]
        }
```

**Review by Rygone — APPROVED** [review]

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/EditProfileActivity.kt (hunk 2 above)
Comment (AlessioVerardo) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/activity/EditProfileActivity.kt]: Yes !

File: app/src/main/res/layout/activity_edit_profile.xml (hunk 1 above)
Comment (AlessioVerardo) [reply app/src/main/res/layout/activity_edit_profile.xml:78]: Yes, it's needed otherwise, the XML gave warning about symmetry :)

**Pull request merged**